docs/branching/new_branching_models.md

```diff
@@ -0,0 +1,111 @@
+
+By using this branching model and implementing best practices as pre merge hooks, you ensure the master lake is never compromised.
+
+<img src="../assets/img/branching_5.png" alt="branching_5" width="500px"/>
+
+
+## Continuous Deployment
+Not every day we introduce new data to the lake, or add/change ETLs, but we do have recurring jobs that are running, and updates to our existing data collections. Even if  the code and infra didn't change, the data might, and those changes introduce quality issues. This is one of the complexities of a data product, the data we consume changes over the course of a month, a week, or even a single day. 
+
+**Examples of changes to data that may occur:**
+ -  A client side bug in the data collection of website events
+ -  A new Android version that interferes with the collecting events from your App
+ -  COVID-19 abrupt impact on consumers behavior , and its effect on the accuracy of ML models.
+ - During a change to Salesforce interface, the validation requirement from a certain field had been lost
+
+lakeFS helps you validate your expectations and assumptions from the data itself.
+
+
+### Example 1: Pre merge hook - a data quality issue
+
+Continuous deployment of existing data we expect to consume, flowing from our ingest-pipelines into the lake. Similar to the Continuous Integration use-case   - we create an ingest branch (“events-data”), which allows us to create tests using data analysis tools or data quality services (e.g. [Great Expectations](https://greatexpectations.io/){: target="_blank" }, [Monte Carlo](https://www.montecarlodata.com/){: target="_blank" }) to ensure reliability of the data we merge to the master branch. Since merge is atomic, no performance issue will be introduced by using lakeFS, but your master branch will only include quality data. 
+
+<img src="../assets/img/branching_6.png" alt="branching_6" width="500px"/>
+
+### Example 2: RollBack! - Data ingested from a Kafka stream
+
+If you introduce a new code version to production and discover  it has a critical bug, you can simply roll back to the previous version. But you also need to roll back the results of running it.  lakeFS gives you the power to rollback your data if you introduced low quality data. The rollback is an atomic action that prevents the data consumers from receiving low quality data until the issue is resolved.
+
+As previously mentioned, with lakeFS the recommended branching schema is to ingest data to a dedicated branch. When streaming data, we can decide to merge the incoming data to master at a given time interval or checkpoint, depending on how we chose to write it from Kafka. 
+
+You can run quality tests for each merge (as presented in Example 1). Alas, tests are not perfect and we might still introduce low quality data at some point. In such a case, we can rollback master to the last known high quality commit, since our commits for streaming will include the metadata of the kafka offset. 
```

Comment:
```suggestion
You can run quality tests for each merge (as presented in Example 1). Alas, tests are not perfect and we might still introduce low quality data at some point. In such a case, we can rollback master to the last known high quality commit, since our commits for streaming will include the metadata of the Kafka offset. 
```